docs/config/templates/app/indexPage.template.html

```diff
@@ -9,6 +9,8 @@
                  Declarative templates with data-binding, MVC, dependency injection and great
                  testability story all implemented with pure client-side JavaScript!">
   <meta name="fragment" content="!">
+  <meta name="robots" ng-attr-content="{{hasError() ? 'noindex' : ''}}" ng-if="hasError()">
+  <meta name="googlebot" content="noindex" ng-if="hasError()">
```

Comment:
Yeah, we I don't believe that this approach works in practice.
We had problems if the raw index.html contained the `noindex` meta tag - removing it dynamically seems too late. We don't know for sure but it seems that if the initial HTML contains this tag the bot doesn't even bother to run the JavaScript.
As you mention @Narretz this commit is the approach we ended up with https://github.com/angular/angular/commit/5a624fa1be530a1b3479a4cc7f96e5a20a3d64fb